spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -355,6 +387,9 @@ def get_matrix_data(
         if variable in self.__per_timestep_variables:
             sampling_rate = 1
             data_type = self.__per_timestep_datatypes[variable]
+        elif variable in self.__events_per_core_variables:
+            sampling_rate = 1
+            data_type = self.__events_per_core_datatypes[variable]
```

Comment:
I think this call won't be made for events-per-core variables as they are handled via the get_rewires function?  Maybe a gate against this, like the above for bitfield variables, is needed.

Reply:
Additionally note (because it won't let me do it directly), there should also be a gate in the get_spikes method in the same way.

Reply:
This isn't needed - a check has been added instead.